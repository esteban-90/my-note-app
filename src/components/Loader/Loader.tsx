import type { FC } from 'react'
import { useTheme } from '@emotion/react'
import { CirclesWithBar } from 'react-loader-spinner'

export const Loader: FC = () => {
  const { colorPalette } = useTheme()

  return (
    <CirclesWithBar
      height='6rem'
      width='6rem'
      ariaLabel='loading indicator'
      wrapperClass='loader'
      barColor={colorPalette.main}
      outerCircleColor={colorPalette.text.accent1}
      innerCircleColor={colorPalette.text.accent2}
    />
  )
}
