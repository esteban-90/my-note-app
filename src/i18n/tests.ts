import type { Resource } from 'i18next'
import { createInstance } from 'i18next'
import { initReactI18next } from 'react-i18next'
import { ns, supportedLngs } from './app'

const resources: Resource = {}
export const i18nInstance = createInstance()

void (async () => {
  for (const n of ns) {
    for (const lng of supportedLngs) {
      resources[lng] ??= {}
      resources[lng] = {
        ...resources[lng],
        [n]: await import(`../../public/translations/${lng}/${n}.json`),
      }
    }
  }
})()

i18nInstance.use(initReactI18next).init({
  ns,
  supportedLngs,
  fallbackLng: supportedLngs[0],
  lng: supportedLngs[0],
  react: { useSuspense: false },
  resources,
})
