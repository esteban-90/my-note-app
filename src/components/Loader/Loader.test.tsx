import { render, setName } from '@/tests'
import { Loader } from './Loader'

describe(setName(Loader), () => {
  let asFragment: () => DocumentFragment

  beforeEach(() => {
    void ({ asFragment } = render(<Loader />))
  })

  it('should match snapshot', () => {
    expect(asFragment()).toMatchSnapshot()
  })
})
