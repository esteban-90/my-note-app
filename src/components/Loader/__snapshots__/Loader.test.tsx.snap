// Vitest Snapshot v1

exports[`LOADER: > should match snapshot 1`] = `
<DocumentFragment>
  <div
    aria-busy="true"
    aria-label="loading indicator"
    class="loader"
    data-testid="circles-with-bar-wrapper"
    role="status"
    style="display: flex;"
  >
    <svg
      data-testid="circles-with-bar-svg"
      enable-background="new 0 0 100 100"
      height="6rem"
      id="L1"
      version="1.1"
      viewBox="0 0 100 100"
      width="6rem"
      x="0px"
      xml:space="preserve"
      xmlns="http://www.w3.org/2000/svg"
      y="0px"
    >
      <circle
        cx="50"
        cy="50"
        fill="none"
        r="47"
        stroke="#acb4bd"
        stroke-dasharray="14.2472,14.2472"
        stroke-miterlimit="15"
        stroke-width="6"
      >
        <animatetransform
          attributeName="transform"
          attributeType="XML"
          dur="5s"
          from="0 50 50"
          repeatCount="indefinite"
          to="360 50 50"
          type="rotate"
        />
      </circle>
      <circle
        cx="50"
        cy="50"
        fill="none"
        r="39"
        stroke="#999"
        stroke-dasharray="10,10"
        stroke-miterlimit="10"
        stroke-width="1"
      >
        <animatetransform
          attributeName="transform"
          attributeType="XML"
          dur="5s"
          from="0 50 50"
          repeatCount="indefinite"
          to="-360 50 50"
          type="rotate"
        />
      </circle>
      <g
        data-testid="circles-with-bar-svg-bar"
        fill="#f68657"
      >
        <rect
          height="30"
          width="5"
          x="30"
          y="35"
        >
          <animatetransform
            attributeName="transform"
            begin="0.1"
            dur="1s"
            repeatCount="indefinite"
            type="translate"
            values="0 5 ; 0 -5; 0 5"
          />
        </rect>
        <rect
          height="30"
          width="5"
          x="40"
          y="35"
        >
          <animatetransform
            attributeName="transform"
            begin="0.2"
            dur="1s"
            repeatCount="indefinite"
            type="translate"
            values="0 5 ; 0 -5; 0 5"
          />
        </rect>
        <rect
          height="30"
          width="5"
          x="50"
          y="35"
        >
          <animatetransform
            attributeName="transform"
            begin="0.3"
            dur="1s"
            repeatCount="indefinite"
            type="translate"
            values="0 5 ; 0 -5; 0 5"
          />
        </rect>
        <rect
          height="30"
          width="5"
          x="60"
          y="35"
        >
          <animatetransform
            attributeName="transform"
            begin="0.4"
            dur="1s"
            repeatCount="indefinite"
            type="translate"
            values="0 5 ; 0 -5; 0 5"
          />
        </rect>
        <rect
          height="30"
          width="5"
          x="70"
          y="35"
        >
          <animatetransform
            attributeName="transform"
            begin="0.5"
            dur="1s"
            repeatCount="indefinite"
            type="translate"
            values="0 5 ; 0 -5; 0 5"
          />
        </rect>
      </g>
    </svg>
  </div>
</DocumentFragment>
`;
